streaming/stream/dir_conf.py

```diff
@@ -0,0 +1,469 @@
+        if not isinstance(validate_hash, Auto):
+            self.check_hashes = _get_hash_algos(validate_hash)
+
+        # 7. Maybe override phaser (phase keeper), then cache self.keep_zip in case needed.
+        self.init_keep_phases = keep_phases
+        self.init_keep_zip = kwargs.get('keep_zip')
+        self.phaser: Phaser
+        self.safe_phaser: Phaser
+
+    def apply_defaults(
+        self,
+        *,
+        split: Union[None, str, Auto] = Auto(),
+        allow_schema_mismatch: Union[bool, Auto] = Auto(),
+        allow_unsafe_types: Union[bool, Auto] = Auto(),
+        allow_unchecked_resumption: Union[bool, Auto] = Auto(),
+        download_retry: Union[str, int, Auto] = Auto(),
+        download_timeout: Union[None, str, float, Auto] = Auto(),
+        download_max_size: Union[None, str, int, Auto] = Auto(),
+        validate_hash: Union[None, str, Sequence[str], Auto] = Auto(),
+        keep_phases: Union[None, str, Sequence[str], Dict[str, bool], Phaser] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Inherit from our owning StreamingDataset the values of any arguments left as auto.
+
+        Args:
+            remote (str, optional): Remote path to stream the dataset from. If ``None``, dataset
+                must be complete locally. Defaults to ``None``.
+            local (str, optional): Local working directory to stream the dataset to. Uses a
+                deterministically-calculated temp directory if not set. Defaults to ``None``.
+            split (str | Auto, optional): Which dataset split sub-path to use, if any. Set to
+                ``Auto`` to not inherit from StreamingDataset. Defaults to ``Auto()``.
+            allow_schema_mismatch (bool | Auto): If ``True``, continue if schemas mismatch across
+                shards, streams, or the whole dataset. If ``False``, raises if schemas mismatch.
+                Set to ``Auto`` to not inherit from StreamingDataset. Defaults to ``Auto()``.
+            allow_unsafe_types (bool | Auto): If ``True``, continue if unsafe type(s) are
+                encountered in shard(s). If ``False``, raises if unsafe type(s) encountered. Set to
+                ``Auto`` to not inherit from StreamingDataset. Defaults to ``Auto()``.
+            allow_unchecked_resumption (bool | Auto): If ``True``, upon resume, accept and use
+                shard file phases that we are unable to check the size/hash(es) of. If ``False``,
+                upon resume, drop such files, to regenerate on the fly when needed. Set to ``Auto``
+                to not inherit from StreamingDataset. Defaults to ``Auto()``.
+            download_retry (stgr | int | Auto): Number of download re-attempts before raising an
+                error. Set to ``Auto`` to not inherit from StreamingDataset. Defaults to
+                ``Auto()``.
+            download_timeout (None | str | float | Auto): Time to wait for a file download to
```

Comment:
Sorry if I misunderstood the logic here; if the user does not provide the `download_timeout`, what would be the default value? Our current (streaming v0.7.4) default value is `60` seconds.

Reply:
The `Auto` jive lets us define the default value in exactly one place (in dataset.py line 254)

I upped `download_timeout` to `'2m'` in light of:
1. All the yamls I see specifically raising it to several to a handful of minutes
2. Assuming our users have network connections slower than us, or as fast as us, but not faster, so if anything we should be raising it for their UX and setting it lower for ourselves
3. Assuming users can tolerate waiting a bit longer in the case of download timeout errors
4. Assuming the incoming Parquet indexing PR which will have to download potentially large Parquet shards will use the same download timeout, because otherwise we would be adding args etc

Reply:
In the future, I wonder if it would make more sense to create a `streaming/defaults.py` or use `streaming/constant(s).py` for such definitions...